### DISCUSSION TITLE
Alternatives to timer task

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#16541] [Q&A]
Hi,

The timer task always seems to give us issues, especially after an app restart. 

For example, it would generate multiple blocking activities, and I noticed the tasks seems to run consecutively vs concurrently.

Does anyone have any advice to implement a better timer task? Open to ideas and suggestions, I just require a task to run jobs on a schedule.

**Comment by Piedone** [comment]
See https://github.com/OrchardCMS/OrchardCore/issues/760#issuecomment-2062898014

However, what are the "blocking activities" you see? Background tasks shouldn't block.

**Reply by weirdyang** [reply]
timer tasks when used in a workflow creates blocking activities,

**Reply by Piedone** [reply]
Those shouldn't block in the async-sync sense though, if that's what you mean, it's just that activities hooked onto a Timer task will wait for that timer to pass. This is desired, of course, and shouldn't cause any issues. Can you elaborate what kind of issues do you see, exactly?

**Reply by weirdyang** [reply]
https://github.com/OrchardCMS/OrchardCore/blob/8d403cda32800723a62d4d100474293994fd0316/src/OrchardCore/OrchardCore.Workflows.Abstractions/Models/Workflow.cs#L48

https://github.com/OrchardCMS/OrchardCore/blob/8d403cda32800723a62d4d100474293994fd0316/src/OrchardCore.Modules/OrchardCore.Workflows/Indexes/WorkflowIndexProvider.cs#L17

Workflow blocking activities, hope that clarifies.

similar to this:
https://github.com/OrchardCMS/OrchardCore/issues/13302
https://github.com/OrchardCMS/OrchardCore/issues/13344

or it just generates a blocking activity that never gets unblocked/resumed. and the only way we can resolve it was by clearing the blocking activities table.
